pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -927,8 +929,9 @@ let
       # i686 issues: https://github.com/NixOS/nixpkgs/pull/117961#issuecomment-812106375
       useZstd = stdenv.buildPlatform.is64bit && versionAtLeast version "5.9";
     in {
-      KERNEL_XZ            = mkIf (!useZstd) yes;
-      KERNEL_ZSTD          = mkIf useZstd yes;
+      KERNEL_UNCOMPRESSED  = mkIf stdenv.hostPlatform.isRiscV yes;
```

Comment:
A comment on the rationale for riscv may be useful here.

Reply:
Added.

Reply:
Done.